treeherder/services/signatures_remover.py

```diff
@@ -0,0 +1,145 @@
+RECEIVER_EMAIL = "dhunt@mozilla.com"
+
+
+class PublicSignaturesRemover:
+    """
+    This class handles the removal of signatures which are (no longer)
+    associated with a job and sends email notifications to the entire team.
+    """
+
+    table_description = """__Summary of recently deleted Performance Signatures__
+> The signatures below were deleted due to the fact that they were no longer associated with a job.
+---
+    """
+
+    table_headers = """
+| Repository | Framework | Platform | Suite | Application |
+| :---: | :---: | :---: | :---: | :---: |
+    """
+
+    def __init__(
+        self,
+        timer: MaxRuntime,
+        taskcluster_model: TaskclusterModel,
+        nr_of_rows_allowed=None,
+        nr_of_emails_allowed=None,
+    ):
+        self.tc_model = taskcluster_model
+        self._subject = "Summary of deleted Performance Signatures"
+        self._content = None
+        self._nr_of_rows_allowed = nr_of_rows_allowed or 50
+        self._nr_of_emails_allowed = nr_of_emails_allowed or 10
+
+        self.timer = timer
+
+    def remove_in_chunks(self, max_timestamp):
+        emails_sent = 0
+        rows_left = self._nr_of_rows_allowed
+        chunk_of_signatures = []
+
+        for performance_signature in PerformanceSignature.objects.filter(
+            last_updated__lte=max_timestamp
+        ):
+            self.timer.quit_on_timeout()
+
+            if emails_sent < self._nr_of_emails_allowed and (
+                not PerformanceDatum.objects.filter(
+                    repository_id=performance_signature.repository_id,  # leverages (repository, signature) compound index
+                    signature_id=performance_signature.id,
+                ).exists()
+            ):
+                rows_left -= 1
+                chunk_of_signatures.append(performance_signature)
+
+                if rows_left == 0:
+                    # extract the proprieties of interest from signatures in a list of dictionaries
+                    email_content = self.__extract_properties(chunk_of_signatures)
+                    # check if Taskcluster Notify Service is up
+                    try:
+                        self.tc_model.notify.ping()
+                    except Exception:
+                        logger.warning("Taskcluster Notify Service is not available")
+                        logger.warning("Failed to delete performance signatures")
+                        break
+                    else:
+                        self._delete(chunk_of_signatures)
+                        self._send_notification(email_content)
+
+                    emails_sent += 1
+                    chunk_of_signatures = []
+                    rows_left = self._nr_of_rows_allowed
+
+        if emails_sent < self._nr_of_emails_allowed and chunk_of_signatures != []:
+            # extract the proprieties of interest from signatures in a list of dictionaries
+            email_content = self.__extract_properties(chunk_of_signatures)
```

Comment:
`email_content` is misleading (I had the impression this is the compiled email). How about `email_data`?